tests/grid/grid_generator_marching_cube_algorithm_02.with_p4est=true.mpirun=1.output

```diff
@@ -0,0 +1,791 @@
+
+DEAL:0::dim=1 iso level: -5.000000e-02
+DEAL:0::point found: -7.000000e-01
+DEAL:0::point found: 7.000000e-01
+DEAL:0::dim=1 iso level: 0.000000e+00
+DEAL:0::point found: -7.500000e-01
+DEAL:0::point found: -7.500000e-01
+DEAL:0::point found: 7.500000e-01
+DEAL:0::point found: 7.500000e-01
```

Comment:
In case the function cuts an element node, we store the value twice within `vertices`. @peterrum do you think we should merge the duplicate vertices?

Reply:
Yes that would make sense. These lines could help you: https://github.com/dealii/dealii/blob/29c3f11d0569fbb5bf13a962fac82c8c07492bf9/source/grid/tria_description.cc#L156-L169

Reply:
I'll do that as a follow-up.